Component of planet initialization on the map

TheShuEd opened the pull request:
<!-- Please read these guidelines before opening your PR: https://docs.spacestation14.io/en/getting-started/pr-guideline -->
<!-- The text between the arrows are comments - they will not be visible on your PR. -->

## About the PR
Currently, creating planetary maps (europa, and terminal) requires turning the map into a planet while still mapping. As far as I know, this has several negative consequences:
1) Problems with saving the map. After creating a planet, the file becomes much less stable, and editing planetary maps more often causes errors
2) Impossibility of random generation. The planet grain is selected on mapping, and is repeated from time to time when loading the map. 
3) Inability to change biomes quickly. 
(I could generally be wrong on all three counts altogether, haha)

But, I propose a simple and elegant solution. A component that connect on the map entity, and starts the planet creation process during MapInitEvent. This ensures that the planet doesn't interfere with mapping, promises easy biome switching, and procedural environment generation every round.


:cl:
- tweak: Now winter around Europa is different every round

TheShuEd commented:
huh, wat

deltanedas commented:
dev probably shouldnt be a planet to not interfere with testing anything that needs to be in space

TheShuEd commented:
Yes, I agree, and that was going to be removed anyway, but now we need to figure out what the error problem is

TheShuEd commented:
but I noticed that in this implementation the evacuation shuttle flies to the right place, not to a random point on the station, gib half the crew.

TheShuEd commented on Content.Server/Parallax/SpawnMapBiomeSystem.cs:
```diff
@@ -0,0 +1,25 @@
+    {
+        if (!_mapManager.IsMap(map)) return;
+
+        _biome.EnsurePlanet(map, _proto.Index(map.Comp.Biome));
```
![image](https://github.com/space-wizards/space-station-14/assets/96445749/ca560429-19e9-4997-9833-ba01df21406f)

TheShuEd replied:
it doesn't really seem like it, does it?

deltanedas replied:
if you add this component in vv to a map entity does it work as normally

and to be clear it works on map entity not station entity

TheShuEd replied:
And i add it to map entity

TheShuEd replied:
Its works in game in any scenarios, but broke integration test

> if you add this component in vv to a map entity does it work as normally

TheShuEd commented:
there's something I can't figure out. Just adding a BiomeComponent to the map breaks the logic?

TheShuEd commented:
Okay, it's working fine now. It is now a component attached to the station prototype. I removed the planet from Dev, and also cleaned up everything related to the planet in the map file from Europe. Instead, I added a new planet initialization component to Europe. Now the winter around Europa is different every time.

![image](https://github.com/space-wizards/space-station-14/assets/96445749/fbe28608-7f93-40ba-b780-73a0420cc2c8)
and on the mapping it is now back in space, which makes it easier to edit.

metalgearsloth commented on Content.Server/Station/Components/StationBiomeComponent.cs:
```diff
@@ -0,0 +1,20 @@
+using Robust.Shared.Prototypes;
+
+namespace Content.Server.Station.Components;
+
+/// <summary>
+/// allows you to initialize a planet on a specific map at initialization time.
+/// </summary>
+
+[RegisterComponent, Access(typeof(StationBiomeSystem))]
```
```suggestion

/// <summary>
/// Runs EnsurePlanet against the largest grid on Mapinit.
/// </summary>
[RegisterComponent, Access(typeof(StationBiomeSystem))]
```

metalgearsloth commented on Content.Server/Station/Components/StationBiomeComponent.cs:
```diff
@@ -0,0 +1,19 @@
+/// </summary>
+[RegisterComponent, Access(typeof(StationBiomeSystem))]
+public sealed partial class StationBiomeComponent : Component
+{
+    [DataField]
+    public ProtoId<BiomeTemplatePrototype> Biome = "Grasslands";
+
```
```suggestion
{
    [DataField(required: true)]
    public ProtoId<BiomeTemplatePrototype> Biome = "Grasslands";

```
